### ISSUE TITLE
3.1.18 introduced error `true === false`

### ISSUE THREAD

**Issue by kentcdodds** [babel/babel-eslint#137]
Upgrading from 3.1.17 to 3.1.18 gives me this error on all of my files:

```
  0:0  error  true === false

✖ 1 problem (1 error, 0 warnings)
```

Any ideas? I have no idea what it could be referring to. A typical file of mine looks like this:

```javascript
export default ngModule => {
  ngModule.directive('someName', someName);

  function someName() {
    // some code
  }
};
```

It may also have an `import` statement or two at the top (though that doesn't seem to make a difference with this error appearing...

It could also look like this:

```javascript
const stateUtils = require('stateUtils');

export default require('registerState')(__filename, require('ngCommon'), {
  // some configuration stuff
});
```

This error appears in either case. I haven't identified a file in which this error doesn't appear. Any ideas?

**Comment by hzoo** [comment]
The latest is 3.1.19 but not sure that will change anything

**Comment by kentcdodds** [comment]
Sorry, I should have mentioned. I tried it with 3.1.19 and it didn't work, then I worked backwards to find the version that results in the error, which is 3.1.18.

**Comment by hzoo** [comment]
Ah ok. Will have to test myself - @PiPeep any ideas?

**Comment by PiPeep** [comment]
@hzoo 3.1.18 also introduced [this commit](https://github.com/babel/babel-eslint/commit/07782378c365d44f67a3ee0a0ef67ff6305b8643), which despite the commit name updated babel to 3.6.4, which I think is far more likely to be causing this issue.

**Comment by PiPeep** [comment]
@kentcdodds: I've tried to reproduce this in a fresh directory, using your example code, but can't. Can you share your `.eslintrc` file? It might be a specific rule that's causing the issue.

**Comment by kentcdodds** [comment]
I just exploded my `.eslintrc` file with a ton of rules, but I'll paste it in here anyway. Hopefully it's helpful!

```
{
  "parser": "babel-eslint",
  "rules": {
    "quotes": [2, "single", "avoid-escape"],
    "no-use-before-define": [2, "nofunc"],
    "no-unused-expressions": 0,
    "strict": 0,
    "new-cap": [2, {
      "newIsCap": true,
      "capIsNew": true,
      "capIsNewExceptions": [
        "DSCreate", "DSUpdate", "DSSave", "DSDestroy", "DSRefresh", "DSChanges", "DSCompute"
      ]
    }],
    "comma-dangle": 0,
    "max-statements": [2, 30], // TODO bring this down...
    "max-len": [2, 120, 2],
    "max-depth": [2, 4],
    "max-nested-callbacks": [2, 5],
    "indent": [2, 2],
    "brace-style": [2, "1tbs"],
    "spaced-line-comment": 0,

    // possible errors
    "no-trailing-spaces": 0,
    "no-multi-str": 2,
    "space-infix-ops": 2,
    "no-cond-assign": 2,
    "no-console": 2,
    "no-constant-condition": 2,
    "no-control-regex": 2,
    "no-process-env": 0,
    "no-debugger": 2,
    "no-dupe-args": 2,
    "no-dupe-keys": 2,
    "no-duplicate-case": 2,
    "no-empty-character-class": 2,
    "no-empty": 2,
    "no-ex-assign": 2,
    "no-extra-boolean-cast": 0,
    "no-extra-parens": 0,
    "no-extra-semi": 2,
    "no-func-assign": 2,
    "no-inner-declarations": 2,
    "no-invalid-regexp": 2,
    "no-irregular-whitespace": 2,
    "no-negated-in-lhs": 2,
    "no-obj-calls": 2,
    "no-regex-spaces": 2,
    "no-reserved-keys": 0,
    "no-sparse-arrays": 2,
    "no-unreachable": 2,
    "use-isnan": 2,
    "valid-jsdoc": 2,
    "valid-typeof": 2,

    // best practices
    "accessor-pairs": 2,
    "no-var": 2,
    "prefer-const": 2,
    "complexity": [2, 5],
    "consistent-return": 2,
    "curly": 2,
    "default-case": 2,
    "dot-notation": 2,
    "dot-location": [2, "property"],
    "eqeqeq": 2,
    "guard-for-in": 2,
    "no-alert": 2,
    "no-caller": 2,
    "no-div-regex": 2,
    "no-else-return": 0,
    "no-labels": 2,
    "no-eq-null": 2,
    "no-eval": 2,
    "no-extend-native": 2,
    "no-extra-bind": 2,
    "no-fallthrough": 2,
    "no-floating-decimal": 2,
    "no-implied-eval": 2,
    "no-iterator": 2,
    "no-lone-blocks": 2,
    "no-loop-func": 2,
    "no-multi-spaces": 2,
    "no-multi-str": 2,
    "no-native-reassign": 2,
    "no-new-func": 2,
    "no-new-wrappers": 2,
    "no-new": 2,
    "no-octal-escape": 2,
    "no-octal": 2,
    "no-param-reassign": 0,
    "no-process-env": 0,
    "no-proto": 2,
    "no-redeclare": 2,
    "no-return-assign": 2,
    "no-script-url": 2,
    "no-self-compare": 2,
    "no-sequences": 2,
    "no-throw-literal": 2,
    "no-unused-expressions": 0,
    "no-void": 2,
    "no-warning-comments": 0, // TODO... turn this on :P jkjk
    "no-with": 0,
    "radix": 2,
    "vars-on-top": 2,
    "wrap-iife": [2, "inside"],
    "yoda": 2,

    // variables
    "no-catch-shadow": 2,
    "no-delete-var": 2,
    "no-shadow-restricted-names": 2,
    "no-shadow": 2,
    "no-undef-init": 2,
    "no-undef": 2,
    "no-undefined": 0,
    "no-unused-vars": [2, {"args": "after-used"}],

    // stylistic
    // "array-bracket-spacing": [2, "never"] // TODO.. turn this on when we're at v0.24.0
    "camelcase": [2, {"properties": "always"}],
    "comma-spacing": [2, {"before": false, "after": true}],
    "comma-style": [2, "last"],
    "computed-property-spacing": [2, "never"],
    "consistent-this": 0, // Too many use-cases for reassigning "this"
    "eol-last": 2,
    "func-names": 0, // I wish... but naming every anonymous function right now would be a royal pain.
    "func-style": [2, "declaration"],
    "key-spacing": [2, {"beforeColon": false, "afterColon": true}],
    "lines-around-comment": 0,
    "linebreak-style": [2, "unix"],
    "new-parens": 2,
    "newline-after-var": 0,
    "no-array-constructor": 2,
    "no-continue": 2,
    "no-inline-comments": 0,
    "no-lonely-if": 2,
    "no-multiple-empty-lines": 0,
    "no-nested-ternary": 2,
    "no-new-object": 2,
    "no-space-before-semi": 2,
    "no-spaced-func": 2,
    "no-ternary": 0,
    "no-trailing-spaces": [2, {"skipBlankLines": true}],
    "no-underscore-dangle": 0,
    "no-unneeded-ternary": 2,
    "object-curly-spacing": [2, "never"],
    "one-var": [2, {"uninitialized": "always", "initialized": "never"}],
    "operator-assignment": 0, // readability on a case-by-case basis
    "operator-linebreak": [2, "after"],
    "padded-blocks": 0,
    "quote-props": [2, "as-needed"],
    "semi-spacing": [2, {"before": false, "after": true}],
    "semi": [2, "always"],
    "sort-vars": 0,
    "space-after-function-name": [2, "never"],
    "space-after-keywords": [2, "always"],
    "space-before-blocks": [2, "always"],
    "space-before-function-paren": [2, "never"],
    "space-in-brackets": [2, "never"],
    "space-in-parens": [2, "never"],
    "space-return-throw-case": 2,
    "space-unary-ops": [2, { "words": true, "nonwords": false }],
    "spaced-comment": 0,
    "wrap-regex": 0,
//    "constructor-super": 2, // TODO when we upgrade eslint
//    "generator-star-spacing": [2, {"before": false, "after": true}], // TODO when we upgrade eslint
//    "no-this-before-super": 2, // TODO when we upgrade eslint
    "object-shorthand": [2, "properties"], // methods are optional so you can specify a name if you want
    "max-params": [2, 10], // I'll tell you, angular and those injectable functions man... TODO reduce this to 4 or 5
    "no-bitwise": 2,
    "no-plusplus": 0
  },
  "env": {
    "browser": true,
    "es6": true
  },
  "ecmaFeatures": {
    "modules": true
  },
  "globals": {
    "module": false,
    "require": false,
    "__filename": false,
    "process": false,
    "ATAC_CONFIG": false
  }
}
```

**Comment by hzoo** [comment]
@PiPeep Ah that's my bad - should be more specific next time.

Also what's your ESLint version actually? Maybe try with all 3 updated? (babel, eslint, babel-eslint)?

**Comment by kentcdodds** [comment]
Upgrading to eslint `0.24.0` fixed the problem (whatever it was). I was hadn't upgraded yet because `eslint-loader` hadn't yet updated it's `peerDependencies`. But now that it has, I have the latest of all babel and eslint related dependencies and everything works. My versions now:

```
"babel": "5.6.14",
"babel-core": "5.6.15",
"babel-eslint": "3.1.19",
"babel-loader": "5.2.2",
"eslint": "0.24.0",
"eslint-loader": "0.14.1",
```

**Comment by kentcdodds** [comment]
Thanks for your help!

**Comment by hzoo** [comment]
Ok cool :smile:
